docs/notebooks/deep_gaussian_processes.pct.py

```diff
@@ -0,0 +1,355 @@
+# %% [markdown]
+# # Using deep Gaussian processes with GPflux for Bayesian optimization.
+
+# %%
+import numpy as np
+import tensorflow as tf
+
+np.random.seed(1794)
+tf.random.set_seed(1794)
+tf.keras.backend.set_floatx("float64")
+
+# %% [markdown]
+# ## Describe the problem
+#
+# In this notebook, we show how to use deep Gaussian processes (DGPs) for Bayesian optimization using Trieste and GPflux.
+#
+# In this example, we look to find the minimum value of the two- and five-dimensional [Michalewicz functions](https://www.sfu.ca/~ssurjano/michal.html) over the hypercubes $[0, pi]^2$/$[0, pi]^5$. We compare a two-layer DGP model with GPR, using Thompson sampling for both.
+#
+# The Michalewicz functions are highly non-stationary and have a global minimum that's hard to find, so DGPs might be more suitable than standard Gaussian processes.
+
+# %%
+import gpflow
+from trieste.objectives import (
+    michalewicz_2,
+    michalewicz_5,
+    MICHALEWICZ_2_MINIMUM,
+    MICHALEWICZ_5_MINIMUM,
+)
+from trieste.objectives.utils import mk_observer
+from util.plotting_plotly import plot_function_plotly
+from trieste.space import Box
+from math import pi
+
+function = michalewicz_2
+F_MINIMIZER = MICHALEWICZ_2_MINIMUM
+
+search_space = Box([0, 0], [pi, pi])
```

Comment:
import search space definition as well

Reply:
I'm not entirely sure what you mean here?